modules/core/src/main/scala/com.snowplowanalytics.snowplow.lakes/Metrics.scala

```diff
@@ -34,18 +35,20 @@ object Metrics {
     received: Int,
     bad: Int,
     committed: Int,
+    latency: FiniteDuration,
     processingLatency: Option[FiniteDuration]
   ) extends CommonMetrics.State {
     def toKVMetrics: List[CommonMetrics.KVMetric] =
       List(
         KVMetric.CountReceived(received),
         KVMetric.CountBad(bad),
-        KVMetric.CountCommitted(committed)
+        KVMetric.CountCommitted(committed),
+        KVMetric.Latency(latency)
       ) ++ processingLatency.map(KVMetric.ProcessingLatency(_))
   }
 
   private object State {
-    def empty: State = State(0, 0, 0, None)
+    def empty: State = State(0, 0, 0, Duration.Zero, None)
```

Comment:
Why isn't the `latency` optional like `processing_latency` ?

Reply:
I'll explain but the explanation is not great!!

I implemented this in the snowflake loader first, before I did it for the lake loader.  And there I decided to make it non-optional because... I _think_ it makes sense to say latency is zero if there are no input events.  If there are no events to process, then it is not lagging, so latency is zero.  I decided it was helpful to emit the zero metric to show the app is healthy  (you could argue this differently though).

Then I did it for the lake loader, and I decided to make the implementation identical to the snowflake loader, so non-optional.

But... the lake loader already had a latency metric: the processing latency.  So why had I made that one optional?

The lake loader is unusual because it sinks events only once per 5 minutes.  So the processing latency (time from receiving to sinking) is like this:

```
undefined
undefined
undefined
undefined
300 (ish)
undefined
undefined
undefined
undefined
300 (ish)
```

For processing latency it did not feel right to say latency = 0 in the intervening minutes.  Because it's not really zero if it's still processing something.

So we've ended up with an inconsistency between the two latency metrics!  One is optional and one is not.  For now, I am OK with this.  And I am OK to revisit the decision once we actually start using these metrics for alarms, reporting and debugging.